### PR TITLE
Adds Costume during Pokemon detection

### PR DESCRIPTION
**Pull request by billyjbryant**
Adds Costume during Pokemon detection

## Description
Checks for costume id during Pokemon detection process.  The costume is stored in the `pokemon` table under the field `costume` and is also sent via webhook data as `costume`.  Also added in checking for `costume` and `form` on GymPokemon and storing this detail in the database. 

## Motivation and Context
This fix allows for detection of the special costume for the Pikachu family during events.  Could also be used to change the sprite accordingly when the Costume is detected; however, this is not built in to this PR. 

## How Has This Been Tested?
My changes were tested on a local install of RocketMap and PokeAlarm
Debian Linux, Python 2.7.13

Testing was done to ensure that the Costume field is being captured and stored in the database as well as delivered via webhook. 

## Screenshots (if appropriate):
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/extras/webhooks.md [hunk 1]
```diff
@@ -224,6 +224,7 @@ A `pokemon` event is sent every time RocketMap detects that a Pokémon has spawn
 | `spawn_start`           | The number of seconds past the hour at which the Pokémon appears      |     `911` |
 | `spawn_end`             | The number of seconds past the hour at which the Pokémon disappears   |    `2710` |
 | `gender`                | The Pokémon's gender<sup>1</sup>                                    |       `2` |
+| `costume`		  | The Pokémon's costume id						  |	  `0` |
```
Comment (sebastienvercammen) [inline docs/extras/webhooks.md]: Indentation

File: pogom/models.py [hunk 2]
```diff
@@ -3067,6 +3069,11 @@ def database_migrate(db, old_ver):
                                     null=False, default=datetime.utcnow())),
             migrator.add_column('gym', 'total_cp',
                                 SmallIntegerField(null=False, default=0)))
+    if old_ver < 21:
+        migrate(
+            migrator.add_column('pokemon', 'costume',
+                                SmallIntegerField(null=True, default=0))
```
Comment (sebastienvercammen) [inline pogom/models.py]: Don't set a field to nullable, then give it a non-null default.

**Comment by pogo-excalibur** [comment]
@billyjbryant Any thoughts on rolling Pokemon forms into this as well?
(to make #2337 unnecessary)

**Comment by billyjbryant** [comment]
@pogo-excalibur are you suggesting to merge https://github.com/RocketMap/RocketMap/pull/2337 with this PR?  Seems https://github.com/RocketMap/RocketMap/pull/2337 does quite a lot more than just add form support.

**Comment by pogo-excalibur** [comment]
@billyjbryant All the image/sprite handling has now been moved to #2348.
So I think the only other relevant part of that PR is saving pokemon forms (both on encounter and in gym-info).
I'm happy to go with (and rework) #2337 if you'd prefer to stick to costumes with this PR, but it seemed similar enough that I thought I'd ask.

**Comment by billyjbryant** [comment]
@pogo-excalibur, I'll merge your form changes in with this PR, does doing that close #2337?

**Comment by pogo-excalibur** [comment]
That would be great.

And if you could add both costumes and forms to GymPokemon as well that would be even better.

**Comment by billyjbryant** [comment]
@pogo-excalibur I rolled the GymPokemon.form edits into this PR.

**Comment by billyjbryant** [comment]
@sebastienvercammen please re-review

**Comment by friscoMad** [comment]
You are still missing the GymPokemon part and now you have files unmerged pushed.
Probably you should restart with a clean branch and readd the changes that are not that many.

**Comment by pogo-excalibur** [comment]
@billyjbryant Is this something that you're still interested in merging?

**Comment by billyjbryant** [comment]
Yes; however, while trying to pull in the last updates to develop, I broke it, I will fix it today and push an updated PR.

**Comment by billyjbryant** [comment]
Replaced with #2461

**Pull request closed without merging**